### PR TITLE
MB-6931 Add request cancellation modal

### PR DESCRIPTION
**Pull request by eamahanna**
## Description

This PR adds the RequestCancellationModal component and it's corresponding tests and story. It also adds two new button classes: tertiary and destructive.
Screen Shot 2021-03-05 at 7 37 34 AM

## Reviewer Notes

Are these the correct colors for the button states?

## Setup

To view the component run storybook via `make storybook`
To run the tests, execute `make client_test`

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6931) for this change

### CODE REVIEW

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import Modal, { ModalTitle, ModalClose, ModalActions, connectModal } from 'components/Modal/Modal';
+
+export const RequestShipmentCancellationModal = ({ onClose, onSubmit }) => (
+  <Modal>
+    <ModalClose handleClick={onClose} />
+    <ModalTitle>
+      <h3>Request shipment cancellation</h3>
+    </ModalTitle>
+    <p>
+      Movers will be notified that this shipment should be canceled. They will confirm or deny this request based on
```
Comment (duncan-truss) [inline src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx:14]: It looks like we're using the British spelling in the title and button but the more common Americanized 'canceled' here.  Might be worth asking @transcom/truss-design for clarity.

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    expect(wrapper.find('ModalClose').exists()).toBe(true);
+    expect(wrapper.find('button[data-testid="modalBackButton"]').exists()).toBe(true);
+    expect(wrapper.find('button[type="submit"]').exists()).toBe(true);
+  });
+
+  it('closes the modal when close icon is clicked', () => {
+    const onClose = jest.fn();
+    const wrapper = mount(<RequestShipmentCancellationModal onSubmit={jest.fn()} onClose={onClose} />);
+
+    wrapper.find('button[data-testid="modalCloseButton"]').simulate('click');
```
Comment (duncan-truss) [inline src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx:21]: Are you getting warnings when running `client_test` that this isn't wrapped in an `act()` block?

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  </Modal>
+);
+
+RequestShipmentCancellationModal.propTypes = {
+  onClose: PropTypes.func,
+  onSubmit: PropTypes.func,
+};
+
+RequestShipmentCancellationModal.defaultProps = {
+  onClose: () => {},
```
Comment (duncan-truss) [inline src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx:34]: Is there a usecase where these wouldn't be required?

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx (hunk 1 above)
Comment (eamahanna) [reply src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx:14]: I believe were are using the following spellings:
```
cancel
Spellings we’re using:

cancellation

canceled (not cancelled)

canceling (not cancelling)

Spellings with double l’s aren’t wrong, but Merriam-Webster lists the version we’re using first which means it’s more common. Our goal is consistency. 
```

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx (hunk 2 above)
Comment (eamahanna) [reply src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx:21]: I don't see any warnings. Are there warnings showing up for you?

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx (hunk 3 above)
Comment (eamahanna) [reply src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx:34]: no, I can remove the default props :)

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx (hunk 2 above)
Comment (duncan-truss) [reply src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.test.jsx:21]: Usually I do see warnings but I guess that's only when inside of a form, all good.

File: src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx (hunk 1 above)
Comment (duncan-truss) [reply src/components/Office/RequestShipmentCancellationModal/RequestShipmentCancellationModal.jsx:14]: Lol consistently inconsistent.

**Review by NamibiaLT — APPROVED** [review]
Looks good!

**Comment by NamibiaLT** [comment]
The integration tests aren't passing, run them again from failing before merging. The Happo tests are showing very minimal changes that are not visible, and they can be approved.

**Comment by duncan-truss** [comment]
@eamahanna It looks like there are some small differences with [Zeplin](https://zpl.io/2pwZNv8), mainly 16px margins on top and bottom of the paragraph.

Screen Shot 2021-03-05 at 1 27 03 PM

**Review by duncan-truss — APPROVED** [review]
🎉

**Review by brooksr4 — APPROVED** [review]

**Pull request merged**